src/noctsleepy/cli.py

```diff
@@ -0,0 +1,129 @@
+            f"Invalid time format: {value}. Expected HH:MM format (e.g., 20:00, 08:30)."
+        )
+
+
+class SleepMetricCategory(str, Enum):
+    """Sleep metric categories."""
+
+    sleep_duration = "sleep_duration"
+    sleep_continuity = "sleep_continuity"
+    sleep_timing = "sleep_timing"
+
+
+@app.command(
+    name="compute-metrics",
+    help="Compute sleep metrics from actigraphy data and save results as JSON.",
+    epilog=(
+        "Results are automatically saved as a JSON file "
+        "in the same directory as the input file."
+    ),
+)
+def compute_metrics(
+    input_data: Annotated[
+        pathlib.Path,
+        typer.Argument(
+            exists=True,
+            resolve_path=True,
+            help="Path to the input data file (CSV or Parquet).",
+        ),
+    ],
+    night_start: Annotated[
+        str | None,
+        typer.Option(
+            "--night-start",
+            "-s",
+            help="Start time of the nocturnal interval in HH:MM format. "
+            "If not provided, defaults to 20:00.",
+            callback=lambda value: parse_time(value) if value else None,
+        ),
+    ] = None,
+    night_end: Annotated[
+        str | None,
+        typer.Option(
+            "--night-end",
+            "-e",
+            help="End time of the nocturnal interval in HH:MM format. "
+            "If not provided, defaults to 08:00.",
+            callback=lambda value: parse_time(value) if value else None,
+        ),
+    ] = None,
+    nw_threshold: Annotated[
+        float,
+        typer.Option(
+            "--nw-threshold",
+            "-t",
+            help="Non-wear threshold fraction (0.0-1.0), "
+            "below which a night is considered valid.",
+            min=0.0,
+            max=1.0,
+        ),
+    ] = 0.2,
+    selected_metrics: Annotated[
+        List[SleepMetricCategory] | None,
+        typer.Option(
+            "--metrics",
+            "-m",
+            help="Specific metric categories to compute. "
+            "If not specified, all metrics are computed. "
+            "Multiple categories can be specified by repeating the option. "
+            "E.g., --metrics sleep_duration --metrics sleep_timing. ",
+            case_sensitive=False,
+            show_choices=True,
+        ),
+    ] = None,
+) -> None:
+    """Compute sleep metrics from actigraphy data.
+
+    This command processes actigraphy data and computes various sleep metrics
+    including sleep duration, continuity, and timing measures. Results are
+    saved as a JSON file in the same directory as the input file.
+    """
+    main.compute_sleep_metrics(
+        input_data=input_data,
+        night_start=night_start,  # type: ignore[arg-type] #Covered by parse_time callback
+        night_end=night_end,  # type: ignore[arg-type] #Covered by parse_time callback
+        nw_threshold=nw_threshold,
+        selected_metrics=[metric.value for metric in selected_metrics]
```

Comment:
Is this list comprehension necessary? As SleepMetricCategory is a string enum, it should act as a string, here, no? No need to extract the value? I might be wrong on this one though.

```suggestion
selected_metrics=selected_metrics
```

Reply:
huh yeah I think you might be right, I'll try it. I vaguely recall needing to grab the value from Enum class? But maybe that's incorrect here

Reply:
Also, this is the same pattern I used for the `wristpy` cli so I probably didn't try alternatives